src/Time/Timestamp.hs

```diff
@@ -0,0 +1,42 @@
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE Rank2Types                 #-}
+{-# LANGUAGE ScopedTypeVariables        #-}
+
+module Time.Timestamp
+       ( Timestamp (..)
+       , timeDiff
+       , timeAdd
+       , timeMul
+       , timeDiv
+       ) where
+
+import GHC.Real (denominator, numerator, (%))
+
+import Time.Rational (KnownRat, RatioNat)
+import Time.Units (Time (..))
+
+-- | Similar to 'Time' but has no units and can be negative.
+newtype Timestamp = Timestamp Rational
```

Comment:
What do you think about names `TimeStamp` and `TimeDiff`?
If we rename `Time` to `TimeDiff` it's better to have `TimeStamp` for consistency, I guess. Maybe there's sense in renaming `Time` into `TimeDiff`.